java/src/main/java/ai/rapids/cudf/ColumnView.java

```diff
@@ -0,0 +1,2943 @@
+  protected ColumnView(long address) {
+    this.viewHandle = address;
+    this.type = DType.fromNative(ColumnView.getNativeTypeId(getNativeView()), ColumnView.getNativeTypeScale(getNativeView()));
+    this.rows = ColumnView.getNativeRowCount(getNativeView());
+    this.nullCount = Optional.of((long)ColumnView.getNativeNullCount(getNativeView()));
+  }
+
+  protected static long viewHandleFromColumnHandle(long columnHandle) {
+    try {
+      long res = ColumnVector.getNativeColumnView(columnHandle);
+      return res;
+    } catch (CudfException e) {
+      ColumnVector.deleteCudfColumn(columnHandle);
+    }
+    return 0;
+  }
+
+  /**
+   * USE WITH CAUTION: This method exposes the address of the native cudf::column_view.  This allows
+   * writing custom kernels or other cuda operations on the data.  DO NOT close this column
+   * vector until you are completely done using the native column_view.  DO NOT modify the column in
+   * any way.  This should be treated as a read only data structure. This API is unstable as
+   * the underlying C/C++ API is still not stabilized.  If the underlying data structure
+   * is renamed this API may be replaced.  The underlying data structure can change from release
+   * to release (it is not stable yet) so be sure that your native code is complied against the
+   * exact same version of libcudf as this is released for.
+   */
+  public final long getNativeView() {
+    return viewHandle;
+  }
+
+  public final DType getType() {
+    return type;
+  }
+
+  /**
+   * Returns the choild column view at a given index.
+   * Please note that it is the responsibility of the caller to close this view.
+   * @param childIndex the index of the child
+   * @return a column view
+   */
+  public final ColumnView getChildColumnView(int childIndex) {
+    int numChildren = getNumChildren();
+    assert childIndex < numChildren : "children index should be less than " + numChildren;
+    if (!getType().isNestedType()) {
+      return null;
+    }
+    long childColumnView = ColumnView.getChildCvPointer(viewHandle, childIndex);
+    return new ColumnView(childColumnView);
+  }
+
+  /**
+   * Gets the data buffer for the current column view (viewHandle).
+   * If the type is LIST, STRUCT it returns null.
+   * @return    If the type is LIST, STRUCT or data buffer is empty it returns null,
+   *            else return the data device buffer
+   */
+  public final BaseDeviceMemoryBuffer getData() {
+    return getDataBuffer(viewHandle);
+  }
+
+  public final BaseDeviceMemoryBuffer getOffsets() {
+    return getOffsetsBuffer(viewHandle);
+  }
+
+  public final BaseDeviceMemoryBuffer getValid() {
+    return getValidityBuffer(viewHandle);
+  }
+
+  /**
+   * Returns the number of nulls in the data. Note that this might end up
+   * being a very expensive operation because if the null count is not
+   * known it will be calculated.
+   */
+  public long getNullCount() {
+    return getNativeNullCount(viewHandle);
```

Comment:
We cached the null count in the constructor.  Why are we going through JNI to get it again?